eng/common/pipelines/templates/steps/git-push-changes.yml

```diff
@@ -41,7 +42,7 @@ steps:
 
 - task: PowerShell@2
   displayName: Push changes
-  condition: and(succeeded(), eq(variables['HasChanges'], 'true'))
+  condition: and(succeeded(), or(eq(variables['HasChanges'], 'true'), eq(${{ parameters.PushForNoChanges }}, 'true')))
```

Comment:
I'm not sure I understand why we need this new PushForNoChanges parameter. If there are no changes then this push will essentially be a noop anyway.

Reply:
We only merge main to live.
There is no changes speaking to `git status`, as we only merge the committed stuff onto live.

Reply:
In `live` branch we want to merge. The `git log` is more like:
```
commit 0851d14bd0cbfe8c0f102c92f8ecf4dc21690f12 (HEAD -> live, main)
Author: someA<someA@microsoft.com>
Date:   Mon Aug 29 11:42:13 2022 -0700

    commit msg 1

commit 626724c5c5fd82fc7f4adc9bf972534993db9d87 (origin/live)
Merge: 4a36170c8165 046896c45684
Author: someB<someB@microsoft.com>
Date:   Wed Jul 27 06:01:44 2022 +0800

    commit msg 2
```
The top commit is what we merge from main branch. To our git-push-changes.yml, we do not have any `git diff`, so `HasChanges` doesn't set, and we have noop. However, we expect to push the top commit "0851d14bd0cbfe8c0f102c92f8ecf4dc21690f12" to the target branch.

Reply:
It solved the problem using variable in pipeline as a workaround. I prefer to have this flexibility in git push yaml. 
We can use it for future cases as well.

Reply:
Couldn't we instead use the SkipCheckingForChanges parameter here?
```suggestion
  condition: and(succeeded(), or(eq(variables['HasChanges'], 'true'), eq(${{ parameters.SkipCheckingForChanges }}, 'true')))
```

That name make more sense to me and also accomplishes the goals we want without needing to worry about the HasChanges runtime variable.

Reply:
We do have some yaml file using the `SkipCheckingForChanges` which do not expect to push changes if this set to true. 
It might introduce some prod surprise. I have the similar changes at the first place, but introduce a new parameter instead finally.

Reply:
Talked offline about the changes.
Use workaround in our pipeline to make this work. 
Will work on this later for a better approach.